src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/CustomerRegisterSuccess.php

```diff
@@ -86,4 +87,23 @@ private function createAuthorizationCode(Mage_Customer_Model_Customer $customer)
             $storage->getClientId(), $customer->getId(), 'customer/register'
         );
     }
+
+    /**
+     * Creates a new quote for customerId and sets it to active
+     *
+     * @param int $customerId
+     */
+    private function createNewCustomerQuote($customerId)
+    {
+        $quoteId = Mage::getModel('checkout/cart_api')->create(Mage::app()->getDefaultStoreView());
+        /** @var Mage_Sales_Model_Quote $quote */
+        $quote = Mage::getModel('sales/quote')
+                     ->setStore(Mage::app()->getDefaultStoreView())
+                     ->load($quoteId);
+
+        $quote->getShippingAddress()->setCollectShippingRates(true);
+        $quote->setCustomerId($customerId)
```

Comment:
the data from the customer object will still need to apply to the newly created quote. See my other comment on it. Maybe that code will need to be put in the helper or something.